### PR TITLE
Phone verification proto

### PR DESCRIPTION
**Pull request by vaishnavi-agr**
Designed the proto of phone verification system.

### CODE REVIEW

File: frontend/Phone_Verific.proto [hunk 1]
```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+
+// Define the package name
+package example;
+option go_package = "dummy_server_gencode_path/";
+// Define the service
+service PhoneVerificationService {
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]: This will be `FrontendService`

File: frontend/Phone_Verific.proto [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+   string sub_title=3;
+   string mob_number_heading=4;
+   string input_numb=5;
+   string mob_num_breifing=6;
+   string img1_url=7;
+   string img2_url=8;
+   Button next_button=9;
+}
+
+message Button {
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]: move to a different proto file in another folder UI

File: frontend/Phone_Verific.proto [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    string text = 1;
+    string bg_color = 2;
+    CornerRadius corner_radius = 3;
+}
+
+message CornerRadius {
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]: same as above

File: frontend/Phone_Verific.proto [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    int32 top_left = 1;
+    int32 top_right = 2;
+    int32 bottom_left = 3;
+    int32 bottom_right = 4;
+}
+
+message TriggerPhoneVerificationRequest {
+    string mob_num = 1;
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]:
```suggestion
    string phone_number = 1;
```

File: frontend/Phone_Verific.proto [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+}
+message TriggerPhoneVerificationResponse {
+    ResponseType response_type = 1;
+}
+
+enum ResponseType {
+    OTP_SENT = 0;
+    INVALID_MOBNUM = 1;
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]:
```suggestion
    INVALID_PHONE_NUMBER = 1;
```

File: frontend/Phone_Verific.proto [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+   string input_numb=4;
+   string mob_num_breifing=5;
+   string img1_url=6;
+   string img2_url=7;
+   Button next_button=8;
+}
+
+
+message GetPhoneVerificationScreenResponse {
+   ResponseType response_type=1;
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]: Why do we need `ResponseType` here ? not able to get why is this needed

File: frontend/Phone_Verific.proto [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    // Define the RPC method
+    rpc GetPhoneVerificationScreen(GetPhoneVerificationScreenRequest) returns (GetPhoneVerificationScreenResponse);
+}
+
+// Define the request message
+message GetPhoneVerificationScreenRequest {
+
+   string title=1;
+   string sub_title=2;
+   string mob_number_heading=3;
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]:
```suggestion
   string phone_number_heading=3;
```

File: frontend/Phone_Verific.proto [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+   string input_numb=4;
+   string mob_num_breifing=5;
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]:
```suggestion
   string phone_number_briefing=5;
```

File: frontend/Phone_Verific.proto [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+   string img1_url=6;
+   string img2_url=7;
+   Button next_button=8;
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]:
```suggestion
   Button primary_button=8;
```

File: frontend/Phone_Verific.proto [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+// Define the request message
+message GetPhoneVerificationScreenRequest {
+
+   string title=1;
+   string sub_title=2;
+   string mob_number_heading=3;
+   string input_numb=4;
+   string mob_num_breifing=5;
+   string img1_url=6;
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]: Why do we need two images here ? not able to get the reason

File: frontend/Phone_Verific.proto (hunk 5 above)
Comment (Jha2508) [reply frontend/Phone_Verific.proto]: Also move this to a common enums file in `enum` folder

**Comment by Jha2508** [comment]
Rename the file to `FrontendService.proto`

File: frontend/Phone_Verific.proto [hunk 11]
```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+
+// Define the package name
+package example;
+option go_package = "dummy_server_gencode_path/";
```
Comment (Jha2508) [inline frontend/Phone_Verific.proto]: Change this path to `FrontendService/`

Comment (vaishnavi-agr) [reply frontend/Phone_Verific.proto]: Done

File: frontend/Phone_Verific.proto (hunk 10 above)
Comment (vaishnavi-agr) [reply frontend/Phone_Verific.proto:20]: Group 38
Group 34
Sir there are 2 images one is large and one is small .

File: frontend/Phone_Verific.proto (hunk 9 above)
Comment (vaishnavi-agr) [reply frontend/Phone_Verific.proto]: Done

File: frontend/Phone_Verific.proto (hunk 8 above)
Comment (vaishnavi-agr) [reply frontend/Phone_Verific.proto]: Done

File: frontend/Phone_Verific.proto (hunk 7 above)
Comment (vaishnavi-agr) [reply frontend/Phone_Verific.proto]: Done

File: frontend/Phone_Verific.proto (hunk 5 above)
Comment (vaishnavi-agr) [reply frontend/Phone_Verific.proto]: Done

File: frontend/Phone_Verific.proto (hunk 4 above)
Comment (vaishnavi-agr) [reply frontend/Phone_Verific.proto]: Done

File: frontend/Phone_Verific.proto (hunk 3 above)
Comment (vaishnavi-agr) [reply frontend/Phone_Verific.proto]: Done

File: frontend/Phone_Verific.proto (hunk 2 above)
Comment (vaishnavi-agr) [reply frontend/Phone_Verific.proto]: Done

File: frontend/Phone_Verific.proto (hunk 1 above)
Comment (vaishnavi-agr) [reply frontend/Phone_Verific.proto]: Done

**Review by Jha2508 — APPROVED** [review]
